<?xml version="1.0" encoding="UTF-8"?>
<launch>
<!-- World Parameters-->
<arg name="world" default="terrain"/>
<arg name="paused" default="true"/>
<arg name="use_sim_time" default="true"/>
<arg name="gui" default="true"/>
<arg name="headless" default="false"/>
<arg name="debug" default="false"/>
<!-- Robot Parameters-->
<arg name="model" default="$(find mysim)/urdf/curiosity_mars_rover.xacro"/>
<arg name="x" default="0.0" />
<arg name="y" default="0.0" />
<arg name="z" default="0.9581811927258969" />
<arg name="roll" default="0.004018227984613128"/>
<arg name="pitch" default="0.00295049275600796"/>
<arg name="yaw" default="3.141592653589793" />
<!-- Launch World-->
<include file="$(find gazebo_ros)/launch/empty_world.launch">
<arg name="world_name" value="$(find mysim)/worlds/$(arg world).world"/>
<arg name="debug" value="$(arg debug)" />
<arg name="gui" value="$(arg gui)"/>
<arg name="paused" value="$(arg paused)"/>
<arg name="use_sim_time" value="$(arg use_sim_time)"/>
<arg name="headless" value="$(arg headless)"/>
</include>
<!-- Spawn Robot-->
<param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />
<node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
args="-urdf -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg roll) -P $(arg pitch) -Y $(arg yaw) -model curiosity_mars_rover -param robot_description"/>
<!-- Robot Joints Controller -->
<include file="$(find mysim)/launch/rover_control.launch"/>
<!-- Robot System APIs -->
<include file="$(find mysim)/launch/rover_systems.launch"/>
<!-- Open Rviz -->
<node name="rviz" pkg="rviz" type="rviz" args="-d $(find mysim)/rviz/rover.rviz"/>
</launch>
